json_serializable/lib/src/generator_helper.dart

```diff
@@ -39,8 +39,20 @@ class _GeneratorHelper {
   final StringBuffer _buffer = new StringBuffer();
 
   String get _prefix => '_\$${_element.name}';
-  String get _mixClassName => '${_prefix}SerializerMixin';
-  String get _helpClassName => '${_prefix}JsonMapWrapper';
+  String _mixinClassName(bool withConstraints) =>
```

Comment:
consider making this 2 separate methods (or getters) rather than using a flag argument. The names will be more clear about what is happening than a cryptic "true" or "false" for which you need to see the argument name to understand.